packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -1180,6 +1185,30 @@ function commitWork(current: Fiber | null, finishedWork: Fiber): void {
       if (primaryChildParent !== null) {
         hideOrUnhideAllChildren(primaryChildParent, newDidTimeout);
       }
+
+      // If this boundary just timed out, then it will have a set of thenables.
+      // For each thenable, attach a listener so that when it resolves, React
+      // attempts to re-render the boundary in the primary (pre-timeout) state.
```

Comment:
I thought we wanted to do this work in the passive effect? Is there a reason not to?

Reply:
Only that it seemed like a riskier change. I'll add a follow-up.